contract/tests/test_contract.py

```diff
@@ -169,6 +169,9 @@ def test_contract(self):
         self.assertAlmostEqual(self.inv_line.price_subtotal, 50.0)
         self.assertEqual(self.contract.user_id, self.invoice_monthly.user_id)
 
+    def test_contract_action_preview(self):
+        self.contract.action_preview()
```

Comment:
Please do an assert about the expected result:
```suggestion
        self.contract.action_preview()
        self.assertIn("/my/contracts/", action["url"])
        self.assertIn("access_token=", action["url"])
```